component/http/cache_builder.go

```diff
@@ -0,0 +1,78 @@
+package http
+
+import (
+	"errors"
+	"time"
+
+	"github.com/beatlabs/patron/cache"
+)
+
+// RouteCache is the builder needed to build a cache for the corresponding route
+type RouteCache struct {
+	// cache is the cache implementation to be used
+	cache cache.Cache
```

Comment:
We should use the TTL version of the cache as it will give us the opportunity to use a cache implementation that supports our needs. Following this approach, we can get rid of the instant which will, and should be handled by the underlying cache which makes the implementation even simpler.

Reply:
agree. This would also allow the change mentioned above to be much more simple.
Only concerns there on my side
- we couple the cache with the route , which introduces component cohesion
- we reduce the scope of the issue (to use the patron Cache interface), which reduces the usage of this feature.
But, having said that, i see that it would make the implementation simpler at this stage.